client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -58,34 +58,35 @@ const HostingFeatures = () => {
 	);
 	const hasEnTranslation = useHasEnTranslation();
 
-	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
-	const isTransferInProgress = [
-		transferStates.PENDING,
-		transferStates.ACTIVE,
-		transferStates.PROVISIONED,
-	].includes( transfer.status );
+	const { data: siteTransferData, refetch: refetchSiteTransferData } = useSiteTransferStatusQuery(
+		siteId || undefined
+	);
+	// We have some bug at the backend, when the transfer is complited, but isSiteAtomic is still false, so we need such extra condition
+	const isTransferInProgress =
+		siteTransferData?.isTransferring || siteTransferData?.status === transferStates.COMPLETED;
 
 	useEffect( () => {
 		if ( ! siteId ) {
 			return;
 		}
 
-		if ( transfer.status === transferStates.COMPLETED ) {
-			window.location.href = `/overview/${ siteSlug }`;
-			return;
-		}
-
-		dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		// A user can keep one tab open and start transfer in the second tab
-		// se we always run interval to check the status, to avoid case with rendering "Activate" button on old tabs of a browser
 		const interval = setInterval( () => {
-			dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		}, 10000 );
+			if ( siteTransferData?.status !== transferStates.COMPLETED ) {
+				refetchSiteTransferData();
+			} else {
+				clearInterval( interval );
+				dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+			}
+		}, 3000 );
```

Comment:
Two things:
1. Making use of react-query's `refetchInterval` in `useSiteTransferStatusQuery` seems like a more appealing option. Maybe we would even fix subtle bugs in other places by making our changes within that hook.
2. Setting `refetchOnWindowFocus` to `true` in `useSiteTransferStatusQuery` would also go a long way. The current refetch interval seems quite short IMO, and `refetchOnWindowFocus` is a smart way to refetch only when needed.

Reply:
Why do we need to also trigger the `fetchAtomicTransfer` action at this point, btw? Is it because that data is used on the other hosting overview pages?

Reply:
>Making use of react-query's refetchInterval in useSiteTransferStatusQuery seems like a more appealing option. Maybe we would even fix subtle bugs in other places by making our changes within that hook.

I thought about it and even try to use first time in my life but, I don't know why, it worked non-stable,  and I decided not to spend time at least on it and proceed with stable manual refetching.

>Setting refetchOnWindowFocus to true in useSiteTransferStatusQuery would also go a long way. The current refetch interval seems quite short IMO, and refetchOnWindowFocus is a smart way to refetch only when needed.

I thought about it too and it's really good approach, but it will leave the initial issue in case of transition Simple->Atomic.
Let's imagine that we go with it.
User hase Simple website and open two tabs with "Hosting features" tab. 
In one of them they click on "Activate features", then moves to another tab and see the same button, *since* on focus we have delay to send request and receive response. I understand that it's rare case, but still.

>Why do we need to also trigger the fetchAtomicTransfer action at this point, btw? Is it because that data is used on the other hosting overview pages?

It's necessary only for the case of transition Simple->Atomic. In that case the data is not updated automatically, but it is updated for the case of A4A websites... TBH, I just gave up at that moment and decided to just refresh manually to have always up-to-data data for sure 😅 I will take a look at it now one more time.

Reply:
> refetchInterval

Maybe we need to combine it with the `refetchIntervalInBackground` setting? Could be worth a try.

> refetchOnWindowFocus

Point taken. If we decide to use `refetchInterval`, then I guess we could still set it to `true`. It might decrease the delay slightly in some instances.